sklearn/utils/extmath.py

```diff
@@ -833,3 +833,13 @@ def _deterministic_vector_sign_flip(u):
     signs = np.sign(u[range(u.shape[0]), max_abs_rows])
     u *= signs[:, np.newaxis]
     return u
+
+
+def weighted_median(array, sample_weight):
+    sorted_idx = np.argsort(array)
+    sample_weight = np.asarray(sample_weight)
+    weight_cdf = sample_weight[sorted_idx].cumsum()
+    weighted_percentile = (weight_cdf - sample_weight/2.0) / weight_cdf[-1]
```

Comment:
I would just add a comment stating that it might not be appropriate to test the `median_absolute_error` because there is a high probability of the median being equal for both the weighted and non-weighted case, i.e either one or zero